src/detectors/builtin/sendInLoop.ts

```diff
@@ -43,29 +48,81 @@ export class SendInLoop extends ASTDetector {
   async check(cu: CompilationUnit): Promise<MistiTactWarning[]> {
     const processedLoopIds = new Set<number>();
     const allWarnings: MistiTactWarning[] = [];
+    const astStore = cu.ast;
+    const ctx = this.ctx;
+    const callGraph = new CallGraph(ctx).build(astStore);
+    const functionsCallingSend = new Set<CGNodeId>();
 
-    Array.from(cu.ast.getProgramEntries()).forEach((node) => {
-      forEachStatement(node, (stmt) => {
-        const warnings = this.analyzeStatement(stmt, processedLoopIds);
-        allWarnings.push(...warnings);
-      });
-    });
+    // Identify all functions that contain a send call
```

Comment:
Let's do it once in `CallGraphBuilder` partially implementing #189. You could e.g. add a bitmask field to `Node` containing some information about it.